### PR TITLE
Autogenerate client from swagger json

### PR DESCRIPTION
**Pull request by mraerino**
Focussed on these goals:
- Generate client from `swagger.json`
- Simplify structure of package (breaking change)
  - `models` and `client` directories

Todos:
- [x] Have up-to-date `swagger.json`
  - [x] Take swagger file from blank netbox installation (2.7.10)
  - [ ] ~Patch swagger spec for incorrect definitions / missing parts~
- [ ] Higher-level abstraction code to handle polymorphism

### CODE REVIEW

File: client/circuits/circuits_choices_read_responses.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Code generated by go-swagger; DO NOT EDIT.
+
+// Copyright 2018 The go-netbox Authors.
```
Comment (awlx) [inline client/circuits/circuits_choices_read_responses.go]: Do we need to change this to 2020?

**Comment by kobayashi** [comment]
@mraerino Still working for this? Sorry for late but, I am planning to add netbox v2.7 for client.

**Comment by kobayashi** [comment]
@mraerino Are you still working for this? I would like to add v2.7 supported version.

**Comment by stefanmcshane** [comment]
@kobayashi @mraerino What is the progress on the 2.7 update? Can we also have the versioning tags match, or closely match the netbox versions?

**Comment by mraerino** [comment]
I did the following to update this PR:
- rebase against master
- replace `swagger.json` with vanilla file from https://netboxdemo.com/api/docs/?format=openapi
- regenerate operations and models using [go-swagger](https://github.com/go-swagger/go-swagger)
- fix license, examples etc

I did not make any changes to the swagger.json which is the main goal here.

If anyone is interested in moving this forward, please try out this branch using this:

```bash
go get -u github.com/netbox-community/go-netbox@86b0ffb1e31c476e90023d048a43ae43a794e3b1
```

**Comment by awlx** [comment]
I cherry-picked some of the ideas to master. Are we going to keep this open or close for good?

**Comment by v0ctor** [comment]
Closing, since there is no activity and these changes were added in #143.

**Pull request closed without merging**